theano/link/basic.py

```diff
@@ -262,14 +268,28 @@ class LocalLinker(Linker):
 
     """
 
-    def make_thunk(self, input_storage=None, output_storage=None, storage_map=None):
+    def make_thunk(
+        self,
+        input_storage: Optional[Any] = None,
+        output_storage: None = None,
+        storage_map: Optional[
+            Union[
+                Dict[
+                    Union["TensorVariable", "TensorConstant"],
+                    Union[List[None], List[ndarray]],
+                ],
+                Dict["TensorConstant", List[ndarray]],
+            ]
+        ] = None,
+        **kwargs,
+    ) -> Any:
```

Comment:
Return type is not specific enough

Reply:
Now returning `Tuple[Callable[[], NoReturn], List[Container], List[Container]]`